pypowsybl/rao/impl/rao.py

```diff
@@ -24,45 +23,49 @@ class Rao:
     def __init__(self, handle: _pypowsybl.JavaHandle):
         self._handle = handle
 
-    def set_crac_file_source(self, network: Network, crac_file: Union[str, PathLike]) -> None:
-        crac = io.BytesIO(open(path_to_str(crac_file), "rb").read())
-        self.set_crac_buffer_source(network, crac)
-
-    def set_crac_buffer_source(self, network: Network, crac_source: io.BytesIO) -> None:
-        _pypowsybl.set_crac_source(network._handle, self._handle, crac_source.getbuffer())
-
-    def set_glsk_file_source(self, network: Network, glsk_file: Union[str, PathLike]) -> None:
-        glsk = io.BytesIO(open(path_to_str(glsk_file), "rb").read())
-        self.set_glsk_buffer_source(network, glsk)
-
-    def set_glsk_buffer_source(self, network: Network, glsk_source: io.BytesIO) -> None:
-        _pypowsybl.set_glsk_source(network._handle, self._handle, glsk_source.getbuffer())
-
-    def run(self, network: Network, parameters: Optional[RaoParameters] = None, rao_provider: str = "SearchTreeRao") -> RaoResult:
+    def run(self, network: Network, parameters: Optional[RaoParameters] = None, rao_provider: str = "SearchTreeRao", crac: Optional[Crac] = None, loop_flow_glsk: Optional[Glsk] = None) -> RaoResult:
         """
         Run a rao from a set of input buffers
         """
         if parameters is None:
             parameters = RaoParameters()
 
-        rao_result = _pypowsybl.run_rao(network=network._handle, rao_context=self._handle, parameters=parameters._to_c_parameters(), rao_provider=rao_provider)
-        crac_handle = _pypowsybl.get_crac(self._handle)
-        return RaoResult(rao_result, crac_handle)
+        if crac is not None:
+            self._crac = crac
+        if self._crac is None:
+            raise _pypowsybl.PyPowsyblError("Providing a CRAC is mandatory to run a rao monitoring")
+        if loop_flow_glsk is not None:
+            _pypowsybl.set_loopflow_glsk(self._handle, loop_flow_glsk._handle)
 
-    def run_voltage_monitoring(self, network: Network, rao_result: RaoResult, load_flow_parameters: Optional[LfParameters] = None, provider_str: str = '') -> RaoResult:
+        rao_result = _pypowsybl.run_rao(network=network._handle, crac=self._crac._handle, rao_context=self._handle, parameters=parameters._to_c_parameters(), rao_provider=rao_provider)
+        return RaoResult(rao_result, self._crac._handle)
+
+    def run_voltage_monitoring(self, network: Network, rao_result: RaoResult, load_flow_parameters: Optional[LfParameters] = None, provider_str: str = '', crac: Optional[Crac] = None, monitoring_glsk: Optional[Glsk] = None) -> RaoResult:
```

Comment:
Same comment regarding the CRAC as above:


```suggestion
    def run_voltage_monitoring(self, crac: Crac, network: Network, rao_result: RaoResult, load_flow_parameters: Optional[LfParameters] = None, provider_str: str = '', monitoring_glsk: Optional[Glsk] = None) -> RaoResult:
```